### ISSUE TITLE
Basis guess does not work with custom DFT functional

### ISSUE THREAD

**Issue by susilehtola** [psi4/psi4#2199]
This calculation fails in Psi4 1.3.2
```
molecule {
Ne
}

my_pbe = {
    "name": "my pbe",
    "x_functionals": {"GGA_X_PBE": {}},
    "c_functionals": {"GGA_C_PBE": {}}
}

set basis pcseg-2
set basis_guess true
energy('scf', dft_functional=my_pbe)
```

**Comment by susilehtola** [comment]
Error is
```
Traceback (most recent call last):
  File "/usr/bin/psi4", line 287, in <module>
    exec(content)
  File "<string>", line 29, in <module>
  File "/usr/lib64//python3.9/site-packages/psi4/driver/driver.py", line 556, in energy
    wfn = procedures['energy'][lowername](lowername, molecule=molecule, **kwargs)
  File "/usr/lib64//python3.9/site-packages/psi4/driver/procrouting/proc.py", line 2002, in run_scf
    scf_wfn = scf_helper(name, post_scf=False, **kwargs)
  File "/usr/lib64//python3.9/site-packages/psi4/driver/procrouting/proc.py", line 1272, in scf_helper
    p4util.banner(name.upper())

AttributeError: 'dict' object has no attribute 'upper'

Printing out the relevant lines from the Psithon --> Python processed input file:
        "x_functionals": {"GGA_X_PBE": {}},
        "c_functionals": {"GGA_C_PBE": {}}
    }
    core.set_global_option("BASIS", "pcseg-2")
    core.set_global_option("BASIS_GUESS", "true")
--> energy('scf', dft_functional=my_pbe)
```

**Comment by hokru** [comment]
Fixed in current version with #1880
